app/javascript/js/controllers/resource_search_controller.js

```diff
@@ -0,0 +1,317 @@
+import { Controller } from '@hotwired/stimulus'
+import { Turbo } from '@hotwired/turbo-rails'
+import { autocomplete } from '@algolia/autocomplete-js'
+import { sanitize } from 'dompurify'
+import URI from 'urijs'
+import debouncePromise from '../helpers/debounce_promise'
+
+export default class extends Controller {
+  static targets = [
+    'autocomplete',
+    'button',
+    'hiddenId',
+    'visibleLabel',
+    'clearValue',
+    'clearButton',
+  ]
+
+  debouncedFetch = debouncePromise(fetch, this.searchDebounce)
```

Comment:
Won't `this.searchDebounce` return link to the function, instead of the result of `searchDebounce()`?

I'd advise adding tests